app/src/full/java/io/homeassistant/companion/android/vehicle/HaCarAppService.kt

```diff
@@ -66,13 +71,19 @@ class HaCarAppService : CarAppService() {
             )
 
             override fun onCreateScreen(intent: Intent): Screen {
+                carInfo = carContext.getCarService(CarHardwareManager::class.java).carInfo
+
                 return MainVehicleScreen(
                     carContext,
                     serverManager,
                     serverIdFlow,
                     entityFlow
                 ) { loadEntities(lifecycleScope, it) }
             }
+
+            override fun onNewIntent(intent: Intent) {
+                carContext.getCarService(ScreenManager::class.java).popToRoot()
```

Comment:
This will create a bug to always go back to the main screen, is this intentational?

Reply:
This is used to show the main screen when we receive the intent from the start notification. Is there any problem with this ?

Reply:
I think we may need to filter by the intent action that we expect to avoid changing the screen on accident if we get an unexpected action.

Reply:
I added an intent filter, but realised that in the end this wasn never called (even without the filter), so I just removed this part.